pywhat/regex_identifier.py

```diff
@@ -12,42 +12,46 @@ def __init__(self):
             self.regexes = json.load(myfile)
 
     def check(self, text):
-        matches = []
-        for txt in text:
-            for reg in self.regexes:
-                matched_regex = re.search(reg["Regex"], txt, re.UNICODE)
-
-                if matched_regex:
-                    reg = copy.copy(reg) # necessary, when checking phone
-                                         # numbers from file that may contain
-                                         # non-international numbers
-                    matched = self.clean_text(matched_regex.group(0))
-
-                    if "Phone Number" in reg["Name"]:
-                        number = re.sub(r"[-() ]", "", matched)
-                        codes_path = "Data/phone_codes.json"
-                        codes_fullpath = os.path.join(
-                            os.path.dirname(os.path.abspath(__file__)), codes_path)
-                        with open(codes_fullpath) as f:
-                            codes = json.load(f)
-
-                        locations = []
-                        for code in codes:
-                            if number.startswith(code["dial_code"]):
-                                locations.append(code["name"])
-                        if len(locations):
-                            reg["Description"] = (
-                                "Location(s)"
-                                + ": "
-                                + ", ".join(locations)
-                            )
-
-                    matches.append(
-                        {
-                            "Matched": matched,
-                            "Regex Pattern": reg,
-                        }
-                    )
+        matches = {}
+        for key, value in text.items():
+            for txt in value:

```

Comment:
Can we use better variables here? What is `txt`? Perhaps a type-hint for the function so I actually know what's in `text.items()`? 😄